### DISCUSSION TITLE
What should I do when an installation requires a restart?

### DISCUSSION THREAD

**Discussion by iloudaros** [The-Wineskin-Project/WineskinServer#241] [Q&A] [answered]
I really don't know if I should be posting questions here. If this is an inappropriate way of asking or help, I am sorry in advance 😅

Hey there and sorry for the noob question, but I am stuck. I am trying to create a wrapper for Microchip Studio. PlayonMac, Porting Kit and WineskinServer seem to have the same problem. They load the installer, I follow the steps in the installer and then it says it requires a restart. I then search for the executable inside drive c and can't find it :/. Any ideas ?

In WineskinServer I am using Wineskin 2.9.0.9-4 and WS11WineCX21.1.0

**Comment by iloudaros** [comment]
That's what I am trying to install https://www.microchip.com/en-us/tools-resources/develop/microchip-studio

**Reply by Gcenx** [reply]
PlayOnMac, PortingKit ( uses Wineskin) are all just frontends to wine on modern versions of macOS all are using something compiled from crossover-sources.

Those kind of installers are rather annoying as they expect windows to have rebooted, I know CrossOver contains its own hacks to fake a reboot but that's not included in their source releases unfortunately.

The only way I can think of possibly making this work would be launching "Winehq Engine Launch" so having at least two Terminal sessions open and running the installer manually in one and once the reboot is required running the command `wine wineboot -r` in the other and hoping that does the job.

However I can't guarantee that would even work nor if the application once installed would even function as intended within wine as I see mentions of accessing devices via USB and it also seems to install an IDE not sure those even run well within wine.

**Reply by iloudaros** [reply]
I see, thank you for taking the time to answer. Crossover does a worse job unfortunately. It doesn't even succeed in running the installer. I ll try a vm using utm.

Have a nice day!

**Comment by Gcenx** [comment, marked as answer]
According to Winehq you should use `wine wineboot -r ` that doesn’t mean this will work thought.

(An explanation to make as the answer)
